plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -153,29 +156,38 @@ public static CachingHiveMetastore memoizeMetastore(HiveMetastore delegate, long
         private HiveMetastore delegate;
         private Optional<Executor> executor = Optional.empty();
         private OptionalLong expiresAfterWriteMillis = OptionalLong.empty();
+        private OptionalLong statsExpiresAfterWriteMillis = OptionalLong.empty();
         private OptionalLong refreshMills = OptionalLong.empty();
         private Long maximumSize;
         private StatsRecording statsRecording = StatsRecording.ENABLED;
         private boolean partitionCacheEnabled = true;
+        private boolean metadataCacheEnabled = true;
+        private boolean statsCacheEnabled = true;
 
         public CachingHiveMetastoreBuilder() {}
 
         private CachingHiveMetastoreBuilder(
                 HiveMetastore delegate,
                 Optional<Executor> executor,
                 OptionalLong expiresAfterWriteMillis,
+                OptionalLong statsExpiresAfterWriteMillis,
                 OptionalLong refreshMills,
                 Long maximumSize,
                 StatsRecording statsRecording,
-                boolean partitionCacheEnabled)
+                boolean partitionCacheEnabled,
+                boolean metadataCacheEnabled,
+                boolean statsCacheEnabled)
         {
             this.delegate = delegate;
             this.executor = executor;
             this.expiresAfterWriteMillis = expiresAfterWriteMillis;
+            this.statsExpiresAfterWriteMillis = statsExpiresAfterWriteMillis;
             this.refreshMills = refreshMills;
             this.maximumSize = maximumSize;
             this.statsRecording = statsRecording;
             this.partitionCacheEnabled = partitionCacheEnabled;
+            this.metadataCacheEnabled = metadataCacheEnabled;
```

Comment:
partitionCacheEnabled should come after metadataCacheEnabled, since partitionCacheEnabled is more granular (subset of) metadata caching

(here & everywhere)